20240726-project-transfer.md

```diff
@@ -0,0 +1,119 @@
+
+### Header
+
+Date: 2024-07-26 
+
+Owner: mobyvb
+
+Accountable:
+- Console team
+
+Consulted:
+-
+
+Informed:
+-
+
+### Context
+
+Users who want to delete their account or transfer billing responsibilities for a project are not currently able to do so.
+
+### Goals
+
+* Allow one user to transfer a project's ownership from themselves to another user
+  - support transferring project billing responsibilities
+  - support deleting an account while preserving project data
+
+### Approach / Design
+
+#### Prerequisites for allowing project transfer
+
+These items must all be true for a project transfer to be processed:
+* There are no unpaid invoices on either the current or new owners' accounts
+  - if current owner has unpaid invoices, it might mean there is usage for this project that we have not been paid for
+  - if new owner has unpaid invoices, it might mean the project is being transferred to a user who won't pay for usage
+* new owner is paid tier (it is okay if current owner is free tier)
+* new owner has not hit their project count limit
+* current owner verifies via email that they approve the transfer of ownership
+* new owner verifies via email that they approve the transfer of ownership, including billing responsibilities for the project
+* neither user is frozen
+
+#### Results of transfer:
+
+* `project.owner_id` changes to new owner's user ID
+* project member record for new owner is created or updated with member type "admin"
+* old owner remains as project member with type "admin"
+
+#### Satellite UI
+
+The project owner will navigate to the "project settings" page in the satellite UI, and interact with a "transfer project" card (TODO designs)
+
+UI flow (current owner):
+
+* click "transfer project"
+* if any of the owner prerequisits fail (unpaid invoices or user is frozen), display a message to the user informing them about what they need to do to transfer a project
+* if prerequisites for owner pass, dialog is displayed asking for the email of the account which should become the new project owner - click "continue"
+* send owner a 6-digit code via email, which they input into the dialog in the UI - click "continue"
+* display success message informing the owner that an email has been sent to the new owner with instructions for completing the transfer
+
+UI flow (new owner):
+
+* on all projects dashboard, the project being transferred appears with some action like "complete transfer" (needs design)
+* if any of the transferee prerequisites fail (unpaid invoices, project limit, free tier, or user is frozen), display a message to the user informing them about what they need to do to complete the transfer
+* if transferee prerequisites pass, dialog is displayed informing the user that by completing the transfer, they will be responsible for paying for usage on the project - click "confirm/continue"
+* send transferee a 6-digit code via email, which they input into the dialog in the UI - click "continue"
```

Comment:
Are the 6-digit codes different between new/old owner?

Reply:
Yes. Added clarifying text.